File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        ans = [0] * len(temperatures)
        for index, temperature in enumerate(temperatures):
            while stack and temperatures[stack[-1]] < temperature:
                prevIndex = stack.pop()
                ans[prevIndex] = index - prevIndex
            stack.append(index)
        
        while stack:
            ans[stack.pop()] = 0
        return ans

File: Data Structures & Algorithms/validate-parentheses/submission-1.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> openBrackets;
        for (char c:s) {
            switch (c) {
                case '[':
                case '{':
                case '(':
                    openBrackets.push(c);
                    continue;
                case ']':
                    if (openBrackets.empty() || openBrackets.top() != '[') return false;
                    break;
                case ')':
                    if (openBrackets.empty() || openBrackets.top() != '(') return false;
                    break;
                case '}':
                    if (openBrackets.empty() || openBrackets.top() != '{') return false;
                    break;
            }
            openBrackets.pop();
        }
        return openBrackets.empty();
    }
};
